src/main/java/hu/flowacademy/zetaabsencemanager/service/AuthenticationService.java

```diff
@@ -0,0 +1,40 @@
+package hu.flowacademy.zetaabsencemanager.service;
+
+import hu.flowacademy.zetaabsencemanager.model.Roles;
+import hu.flowacademy.zetaabsencemanager.model.User;
+import hu.flowacademy.zetaabsencemanager.repository.UserRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import org.springframework.web.server.ResponseStatusException;
+
+@Service
+@Transactional
+public class AuthenticationService {
+
+    @Autowired
+    private UserRepository userRepository;
+
+    public User getCurrentUser() {
+        var principal = SecurityContextHolder.getContext().getAuthentication().getPrincipal();
+        return userRepository.findByEmailAndDeletedAtNull(principal.toString()).orElseThrow(() -> new ResponseStatusException(HttpStatus.UNAUTHORIZED, "User not found"));
+
+/*        return Optional.ofNullable(SecurityContextHolder.getContext().getAuthentication())
```

Comment:
Legyen optional-lal megoldva a `SecurityContextHolder.getContext().getAuthentication().getPrincipal()`

Reply:
Meg utána a repo hívás is